### PR TITLE
Add updateAnnotatedFile property

### LINKED ISSUE #131: Generator adds comment into version controlled source file containing the Openapi annotation
### Description of the bug

Every time, when the OpenAPI generator runs, it adds following comment line to the source file containing the Openapi annotation:
```
// Openapi Generator last run: : 2024-02-01T14:45:00.661335
```
Since that source file is typically under source control its automatic modification is very annoying, because a developer has either to revert the git change or commit it after each `dart run build_runner build`. The situation becomes even worse, if the `dart run build_runner build` is executed each time before the normal `flutter build`.

### Steps to reproduce

execute `dart run build_runner build`

### Expected behavior

The source file containing the openapi annotation is not touched.

### Logs

_No response_

### Screenshots

_No response_

### Platform

macOS

### Library version

5.0.2

### Flutter version

3.16.9

### Flutter channel

stable

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by aoisupersix**
Fixes #131

This PR adds `updateAnnotatedFile` property to `@OpenApi()` annotation.

When the timestamp is forcibly added or updated to the `@OpenApi()` annotated class, frequent differences occur in source control. Therefore, by adding the `updateAnnotatedFile` property, if `false` is set, the addition or update of the timestamp will not be performed.

```dart
@Openapi(
  ...
  updateAnnotatedFile: false,
)
```

Default is `true` to keep current behavior.

The operation was checked by uncommenting in `openapi-generator/pubspec.yaml`.

```yaml
# dependency_overrides:
#  openapi_generator_annotations:
#    path: ../openapi-generator-annotations
```

### CODE REVIEW

**Comment by Rick-VA** [comment]
We tested it on 5 APIs, and it worked perfectly.

**Review by Nexushunter — APPROVED** [review]

**Comment by DmitrySboychakov** [comment]
Also interested in this. When this can be merged?

**Comment by tjarvstrand** [comment]
Great change! Please merge :)

**Comment by aoisupersix** [comment]
This PR is blocked because it requires a review from a user with write permissions.
Also the package update process will likely be necessary after merge.
Therefore need to wait for owner (@gibahjoe) to respond.

**Comment by tjarvstrand** [comment]
@aoisupersix Yes, sorry, my comment was meant for @gibahjoe :)

**Comment by gibahjoe** [comment]
I am soo sorry, been busy with studies all year. 

The downside of this PR is that you will need to manually update the file containing the annotation before build runner will pick it up again next time there are changes. But I understand why you need more control.

**Pull request merged**
